config/always-python.m4

```diff
@@ -47,41 +58,42 @@ AC_DEFUN([ZFS_AC_CONFIG_ALWAYS_PYTHON], [
 
 	AS_CASE([$with_python],
 		[check],
-		[AS_IF([test -x /usr/bin/python3],
+		[AS_IF([test -x `which python3`],
```

Comment:
This  is going to need to be quoted, otherwise on Linux it will always return true.  It's possible FreeBSD handles this differently, I'm not sure.

Changing this to `[AS_IF([test -x "$(which python3)"],` worked as expected in my testing and fixed the default with `--with-python=check` functionality to detect the system python..  I'd suggest we also use `$()` here rather than `\` \``.